include/zephyr/arch/arm/aarch32/cortex_a_r/scripts/linker.ld

```diff
@@ -386,8 +386,10 @@ GROUP_END(OCM)
     LINKER_DT_SECTIONS()
 
     /* Must be last in romable region */
-    SECTION_PROLOGUE(.last_section,(NOLOAD),)
+    SECTION_PROLOGUE(.last_section,,)
     {
+	*(.shared_isr_args)
+	*(.shared_isr)
```

Comment:
I'm very very skeptical about this, so you must really convince me that this is the correct way to do this.

1. This is working with `CONFIG_CMAKE_LINKER_GENERATOR=y ` ?
2. What happens to the platforms that do not have a `.last_section`?
3. Why this is (again) AArch32 specific?

Reply:
@carlocaione I'm working on an STM32 project that requires both USB and UCPD but they share IRQ 8. So as it stands, I can't use those peripheral together and my project is on hold. I was mainly focusing on AArch32 because that's the architecture of my microcontroller but I planned to add other architectures to this PR once I got approved changes for the AArch32.

I'm relatively new to Zephyr and have mostly contributed drivers and subsystems but not fundamental changes such as those in this PR. Could you advise on the correct way to implement shared ISRs in Zephyr.

Reply:
FWIW: I'm fine if this is supported by only one arch to start with as long as the API is clean.  The code to manage and interpret the ISR table should be generic and documented, but the instructions that actually invoke the pointers are all in per-architecture interrupt entry layers.

Reply:
I'm fine with having this supported by only one arch to start with, but some trivial (possibly) changes can make this PR easily adoptable by other architectures as well.

Instead right now we have:
- AArch32 specific test while this should really be a general test for all the architectures supporting this because it's not relying on anything special to AArch32. We can of course enable this feature and test only for AArch32 at the beginning.
- The whole mechanism relies on a change to AArch32 linker scripts while we should really make this generic (see 3a48365babbdacda4547a2076061592912b6e474 for inspiration and in general the whole #46769 where we tried to decouple the arch linker scripts from whatever it is generated by the `gen_isr_tables.py` script)
- `CONFIG_CMAKE_LINKER_GENERATOR` is not correctly supported yet. When compiling the test we get:
```
warning: orphan section `.shared_isr' from `zephyr/CMakeFiles/zephyr_final.dir/isr_tables.c.obj' being placed in section `.shared_isr'
warning: orphan section `.shared_isr_args' from `zephyr/CMakeFiles/zephyr_final.dir/isr_tables.c.obj' being placed in section `.shared_isr_args'
```
In general the changes to the PR are slowly converging but IMO this is still a bit of work to do but it's really matter of moving to a standalone test and adding a bunch of `zephyr_linker_sources_ifdef` instead of touching the linker scripts.

Reply:
Thanks for the detailed explanation. I've uploaded a new PR that hopefully moves things in the right direction. General interrupt tests still need to be written. I have not found a way to set a pending interrupt in the tests that works like NVIC_SetPendingIRQ.

Reply:
Also, setting CONFIG_CMAKE_LINKER_GENERATOR=y does not generate build errors.